diff --git a/packages/react/src/components/OverflowMenuItem/OverflowMenuItem-test.js b/packages/react/src/components/OverflowMenuItem/OverflowMenuItem-test.js
--- a/packages/react/src/components/OverflowMenuItem/OverflowMenuItem-test.js
+++ b/packages/react/src/components/OverflowMenuItem/OverflowMenuItem-test.js
@@ -1,5 +1,5 @@
 /**
- * Copyright IBM Corp. 2016, 2023
+ * Copyright IBM Corp. 2016, 2026
  *
  * This source code is licensed under the Apache-2.0 license found in the
  * LICENSE file in the root directory of this source tree.
@@ -101,6 +101,39 @@ describe('OverflowMenuItem - RTL', () => {
       expect(container.firstChild.firstChild).toHaveProperty('disabled', true);
     });
 
+    it('should set aria-disabled on the button node when disabled', () => {
+      render(
+        <OverflowMenuItem closeMenu={jest.fn()} itemText="one" disabled />
+      );
+
+      expect(screen.getByRole('menuitem')).toHaveAttribute(
+        'aria-disabled',
+        'true'
+      );
+    });
+
+    it('should set aria-disabled on the anchor node when disabled', () => {
+      render(
+        <OverflowMenuItem
+          closeMenu={jest.fn()}
+          href="https://carbondesignsystem.com"
+          itemText="one"
+          disabled
+        />
+      );
+
+      expect(screen.getByRole('menuitem')).toHaveAttribute(
+        'aria-disabled',
+        'true'
+      );
+    });
+
+    it('should not set aria-disabled when not disabled', () => {
+      render(<OverflowMenuItem closeMenu={jest.fn()} itemText="one" />);
+
+      expect(screen.getByRole('menuitem')).not.toHaveAttribute('aria-disabled');
+    });
+
     it('should have divider', () => {
       const { container } = render(
         <OverflowMenuItem closeMenu={jest.fn()} itemText="one" hasDivider />

diff --git a/packages/react/src/components/OverflowMenuItem/OverflowMenuItem.tsx b/packages/react/src/components/OverflowMenuItem/OverflowMenuItem.tsx
--- a/packages/react/src/components/OverflowMenuItem/OverflowMenuItem.tsx
+++ b/packages/react/src/components/OverflowMenuItem/OverflowMenuItem.tsx
@@ -176,6 +176,7 @@ const OverflowMenuItem = frFn((props, ref) => {
   return (
     <Text as="li" className={overflowMenuItemClasses} role="none">
       <TagToUse
+        aria-disabled={disabled ? true : undefined}
         className={overflowMenuBtnClasses}
         disabled={disabled}
         href={href}